api/tests/unit/test_models.py

```diff
@@ -33,52 +33,35 @@ def test_new_project(new_project):
     assert new_project.ghg_reduced == 2.234
 
 
-def test_insert_location():
-    app = create_app('test')
-    ctx = app.app_context()
-    ctx.push()
-    db.create_all()
+def test_insert_location(app):
     location = Location(address='123 testing way')
     db.session.add(location)
     db.session.commit()
+    assert location is not None
```

Comment:
I don't think this assert statement checks that a record was created in the database for the new location object. It only checks that the location object was instantiated.

Reply:
I think the way I looked at it was that the test_select_location below it checks if a location is successfully created in the db. And this test_insert_location just tried to catch anything fishy from the instantiation and commit to the db, regardless of if it is actually stored in the db. I'm still developing my intuition for testing so I don't have a strong feeling on this. Let me know what your thinking on this is.

Reply:
Sorry, this was the last comment I had time to make this morning, and I didn't see that you had tested for database inserts further below. As long as we cover that, it probably doesn't matter too much whether we do it with two separate tests or one test. When the test suite grows larger, it might make more sense to consolidate it into one test to minimize the cost of setup/teardown.

Reply:
Yeah that makes sense. Refactoring some of the tests into more general ones will be more and more appropriate as we add more tests.